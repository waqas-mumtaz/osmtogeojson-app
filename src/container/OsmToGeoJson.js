import React, { useState, useEffect } from "react";
import useHttp from "../hooks/http";
import Search from "../components/Search/search";
import Notification from "../components/UI/Notification";
import LoadingIndicator from "../components/UI/LoadingIndicator";
const OsmToGeoJson = () => {
  const [hasError, setErrors] = useState(false);

  //search input
  const [enteredFilter, setEnteredFilter] = useState("");
  const [dataFetched, setDateFetched] = useState("");
  const { isLoading, data, error, sendRequest, clear } = useHttp();

  useEffect(() => {
    if (enteredFilter !== "") {
      sendRequest(enteredFilter, "GET");
    }
  }, [sendRequest, enteredFilter]);

  useEffect(() => {
    if (!isLoading && !error && data) {
      setDateFetched(data);
    }
  }, [data, isLoading, error]);

  return (
    <>
      <section className="section">
        <div className="container">
          <h1 className="title">OSM to GEOJSON</h1>
          <section className="section">
            <div className="columns">
              <div className="column">
                <Search
                  inputValue={(value) => setEnteredFilter(value)}
                  clear={() => setEnteredFilter("")}
                />
              </div>
            </div>
          </section>
          <div className="columns">
            <div className="column">
              {error ? (
                <Notification className="is-danger is-light">
                  {error}
                </Notification>
              ) : isLoading ? (
                <LoadingIndicator />
              ) : (enteredFilter !== "") & (dataFetched !== "") ? (
                <pre>{JSON.stringify(dataFetched, undefined, 3)}</pre>
              ) : (
                <div class="card">
                  <div class="card-content">
                    <div class="content">
                      where:
                      <ul>
                        <li>
                          <code>left</code> is the longitude of the left
                          (westernmost) side of the bounding box.
                        </li>
                        <li>
                          <code>bottom</code> is the latitude of the bottom
                          (southernmost) side of the bounding box.
                        </li>
                        <li>
                          <code>right</code> is the longitude of the right
                          (easternmost) side of the bounding box.
                        </li>
                        <li>
                          <code>top</code> is the latitude of the top
                          (northernmost) side of the bounding box.
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              )}
            </div>
          </div>
        </div>
      </section>
    </>
  );
};

export default OsmToGeoJson;
