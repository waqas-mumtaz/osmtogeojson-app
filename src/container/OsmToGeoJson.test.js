import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import OsmToGeoJson from "./OsmToGeoJson";
import LoadingIndicator from "../components/UI/LoadingIndicator";

configure({ adapter: new Adapter() });

describe("<OsmToGeoJson />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OsmToGeoJson />);
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
