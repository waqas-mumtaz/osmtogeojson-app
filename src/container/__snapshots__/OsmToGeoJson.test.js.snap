// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OsmToGeoJson /> should render correctly 1`] = `ShallowWrapper {}`;
